rfcs/0011-reusable-reactors-folder.md

```diff
@@ -0,0 +1,115 @@
+
+### Supporting reactors
+[supporting-reactors]: #supporting-reactors
+
+According to this proposal, the `lib/` directory is designated for reusable reactors, while the `src/` directory is reserved for executable programs. In some projects, there may be reactors that are used internally by a library but are not intended to be exposed to its users. These can be considered "supporting reactors." To clearly mark such reactors as internal and not for public use, we propose using a `private/` directory within the `lib/` folder. This directory can exist at any level of hierarchy within `lib/` and signifies that its contents are private and should not be exposed to users.
+
+For example:
+
+```shell
+├── root
+│   ├── bin/
+│   ├── include/
+│   ├── lib/
+│   │   ├── Input.lf
+│   │   ├── ComputerVision.lf
+│   │   └── private/
+│   │       ├── SupportingReactor1.lf
+│   │       └── SupportingReactor2.lf
+│   ├── src/
+│   ├── src-gen/
+│   ├── fed-gen/  # Only for federated programs (instead of src-gen)
+│   └── Lingo.toml
+```
+
+In this structure, the `private/` directory contains `SupportingReactor1.lf` and `SupportingReactor2.lf`. These reactors are intended to be extended by other reactors or imported for specific functions, but are not meant to be exposed to users of the library. This convention helps maintain a clear separation between public and internal components of the library.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Here are some potential drawbacks of this proposal:
+- **Lack of flexibility**: The proposed structure is rigid and may not be suitable for all projects. It may require additional modifications or extensions to accommodate specific requirements.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+### Alternative Names for the `lib/` Folder
+The folder containing reusable reactors has been named `lib/` since the idea is to have a centralized location for all reusable reactors that can be imported into other projects, similar to how libraries are used in other programming languages. However, we’re open to discussing different names for this folder if the proposed one causes confusion among developers. Some alternative names that could be considered include:
+- `reactors_lib/` or `lfc_reactors/`
+- `libraries/`, `lfc_libraries/` or `lfc_lib/`
+- `reusables/`
+- `modules/`
+
+### Alternative Design Approach
+Another approach could involve placing the lib/ folder under the src/ directory, as illustrated below:
+```shell
+├── root
+│   ├── ...
+│   ├── src/
+│   │   ├── lib/ # Directory for storing reusable reactors
+│   │   ├── Main.lf
+│   └── ...
+```
+This structure nests the `lib/` folder within `src/`, organizing reusable reactors alongside other source files. This approach may enhance clarity and organization, but it may also lead to potential conflicts or inconsistencies in the codebase.
```

Comment:
```suggestion
This structure nests the `lib/` folder within `src/`, organizing reusable reactors alongside other source files. This approach may enhance clarity and organization, but it may also lead to potential conflicts or inconsistencies in the codebase.
```

Please elaborate on the potential of conflicts or inconsistencies.

Reply:
This remained unanswered during the meeting...